nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/util/SnippetUtils.java

```diff
@@ -1015,4 +1018,33 @@ private void normalizeCoordinates(Collection<? extends ComponentDTO> components)
         }
     }
 
+    public static void stripNonUiRelevantFields(final ControllerServiceEntity serviceEntity) {
```

Comment:
What was the motivation for adding these methods to `SnippetUtils`?

Reply:
It's referenced from a few different places. So I wanted to extract the logic into some sort of utility class that could be easily referenced from all of the resources that needed it. We could create a separate util class  - `ControllerServiceUtils` or something like that maybe? Didn't feel like that warranted its own util class, and i figured one could argue that this deals with a snippet of the flow. Though I suppose this util class is actually intended for interacting with Snippets with a capital S, the specific components. I could move it into its own util class if you prefer.

Reply:
`SnippetUtils` is intended for interacting with Snippets with a capital S. We could do another class or we could even put it in the `ApplicationResource`. I believe all the calling code is in a class that extends `ApplicationResource`.

Reply:
Yeah fair. Could definitely put it in ApplicationResource.